api/v1alpha1/ratelimit_types.go

```diff
@@ -91,6 +92,91 @@ type RateLimitRule struct {
 	// 429 HTTP status code is sent back to the client when
 	// the selected requests have reached the limit.
 	Limit RateLimitValue `json:"limit"`
```

Comment:
It seems like it is still required to specify the number of requests limit, how we can handle the case when we only care of the cost not the number of the request limit?

Reply:
also with this new `cost` field, `limit.requests` doesnt quite fit anymore, something like `limit.total` feels better and can be an alias to `limit.requests` in the future to improve defining intent